### PR TITLE
Fix oom kill unit test

### PR DESCRIPTION
**Pull request by Arun-Niranjan**
Addresses #335 

Changes include:
- adding the option to specify a random seed when generating an agent spec
- adding an option to specify the maximum A dependency depth limit
- using these options in the unit test that was randomly OOM-ing.
- using JAX instead of numpy for the random number generation

~~Default behaviour remains unchanged~~
Note: in keeping with JAX norms, the default behaviour is now changed. Instead of non-deterministic by default, we assume a key of 0 if none is provided. Because these methods are only used in the tests (which passed in a seed before) this is not a breaking change to anything else in pymdp

### CODE REVIEW

File: pymdp/utils.py [hunk 1]
```diff
@@ -365,13 +368,18 @@ def generate_agent_spec(num_factors,
     A_dep_len_prior:   'uniform'| 'exponential' - whether to sample lengths of A
                        dependency lists uniformly or to favor shorter ones, in cases
                        where no hidden states have already been assigned to a list
+    rng:               numpy random Generator or RandomState, for reproducible random
+                       number generation. If None, uses default_rng()
     '''
+
+    if rng is None:
```
Comment (conorheins) [inline pymdp/utils.py]: Can you refactor this to use jax PRNGKeys here instead of numpy random keys? I think that anytime we have an opportunity to change something related to random seeding, we should refactor if possible to using jax.random

File: test/test_infer_states_optimized.py [hunk 2]
```diff
@@ -73,7 +77,9 @@ def setUpClass(cls):
         spec_data = generate_agent_specs_from_parameter_sets(
             parameter_sets,
             num_agents_per_set=1,
-            output_file=None  # Don't save to file
+            max_A_dependency_list_size=3,
```
Comment (conorheins) [inline test/test_infer_states_optimized.py:80]: waiting on confirmation from @nikolamilovic-ft to see whether reducing this to 3 is valid/sufficient for validating the behavior of the optimized inference methods

Comment (Arun-Niranjan) [reply test/test_infer_states_optimized.py:80]: There's a case to be made that if this test *needs* to be using a ton of memory, it should be a benchmark that runs on a slower cadence than the standard unit tests

Comment (Arun-Niranjan) [reply test/test_infer_states_optimized.py:80]: I'm going to leave this as it is for now to ensure we don't have flaky CI - happy to create a ticket for adding memory and speed benchmark tests that can be run in a script locally, or perhaps we fork out for more powerful github action runners that run once a week

Comment (conorheins) [reply test/test_infer_states_optimized.py:80]: Sounds good, got confirmation that the max_A_dependency_list_size=3 is sufficient for establishing that the methods work as intended anyway
